### PR TITLE
gc_worker: update gc_delete_range table should be in txn.

### PR DESCRIPTION
**Pull request by hicqu**

### CODE REVIEW

**Comment by hicqu** [comment]
@hanfei1991 @disksing , PTAL, thanks.

**Comment by disksing** [comment]
LGTM.

File: store/tikv/gc_worker.go [hunk 1]
```diff
@@ -381,8 +381,13 @@ func (w *GCWorker) deleteRanges(ctx goctx.Context, safePoint uint64) error {
 			}
 			startKey = endKey
 		}
-		err := ddl.CompleteDeleteRange(w.session, r)
-		if err != nil {
+		if _, err := w.session.Execute("BEGIN"); err != nil {
+			return errors.Trace(err)
+		}
+		if err := ddl.CompleteDeleteRange(w.session, r); err != nil {
```
Comment (ngaut) [inline store/tikv/gc_worker.go]: Maybe we should add a check in CompleteDeleteRange in case we forget it next time.

**Comment by zimulala** [comment]
Is `autocommit ==1` true? If it's true, I think it will commit automatically.

**Comment by hicqu** [comment]
It's not correct way to fix this. I will start a new PR.

**Pull request closed without merging**